### PR TITLE
tests: Add a big (many objects) pull

### PR DESCRIPTION
**Pull request by cgwalters**
This would be more likely to tickle things like
https://github.com/ostreedev/ostree/issues/601
reliably.

Also, while working on the curl backend, I hit on the fact that curl doesn't
queue (by default, you can enable) and will happily create 20000+ concurrent TCP
connections if you try. Having this test would have made that more likely to
fail.

### CODE REVIEW

**Comment by cgwalters** [comment]
Tangential fun fact: when testing this originally I was running nginx in a Docker container with the default bridge, and this killed `docker-proxy`.  Ended up running nginx with `--net=host` to avoid that.

File: tests/test-pull-many.sh [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+cd ${test_tmpdir}
+rm main -rf
+mkdir main
+cd main
+ndirs=9443
+depth=0
+echo "$(date): Generating content..."
+while [ $ndirs -gt 0 ]; do
+    # 2/3 of the time, recurse a dir, up to a max of 9, otherwise back up
+    x=$(($RANDOM % 3))
```
Comment (giuseppe) [inline tests/test-pull-many.sh]: should we avoid $RANDOM to make the input repeatable?

**Comment by giuseppe** [comment]
tests are still failing, it seems that the `echo "ok ...` message is not present

**Comment by jlebon** [comment]
bot, retest this please

**Comment by giuseppe** [comment]
@rh-atomic-bot r+  	32d7de7

**Comment by cgwalters** [comment]
@giuseppe try again

**Comment by cgwalters** [comment]
@rh-atomic-bot r=giuseppe

**Pull request closed without merging**